app/models/assignment_properties.rb

```diff
@@ -131,4 +133,29 @@ def not_timed_and_scanned
     msg = I18n.t('activerecord.errors.models.assignment_properties.attributes.is_timed.not_scanned')
     errors.add(:base, msg) if is_timed && scanned_exam
   end
+
+  def default_starter_group_not_empty
+    default_sfgroup = StarterFileGroup.find_by(id: default_starter_file_group_id)
+    msg = I18n.t('activerecord.errors.models.assignment_properties.attributes.default_starter_file_group_id.is_empty')
+    errors.add(:base, msg) if default_sfgroup&.starter_file_entries&.empty?
+  end
+
+  # Warn groupings that are affected by a change in starter_file_type. If the type changed to/from 'shuffle' then this
+  # should be all groupings. Otherwise, it only affects groupings that have not been assigned any starter files yet.
```

Comment:
I'm not sure this is correct. For example, if we go from 'simple' to 'section', won't we want to reassign students who are in a section that needs a different starter file group?

Reply:
No, because we prioritize making sure that students who have already received starter code keep the same version of that starter code if at all possible